poetry/console/commands/add.py

```diff
@@ -123,6 +131,9 @@ def handle(self):
             if self.option("platform"):
                 constraint["platform"] = self.option("platform")
 
+            if self.option("source"):
+                constraint["source"] = self.option("source")
```

Comment:
I think we should check if the source is defined in the `pyproject.toml` file before adding it to the constraint.

Reply:
It will fail later with an explicit message (there is a test for it). I find it better than to silently ignore the constraint, WDYT?

Reply:
Ah yes, I didn't see it! You are right.